src/Parsers/ExpressionElementParsers.cpp

```diff
@@ -309,6 +309,44 @@ bool ParserFunction::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
     return true;
 }
 
+bool ParserTableFunction::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
```

Comment:
Maybe just allow to have query as an argument to any function?
And then throw exception on query analysis stage if it's wrong place...

Thus we can avoid having special case for "view" here.